extension/src/popups/MinervaMenu.js

```diff
@@ -1,41 +1,78 @@
-import { renderStudentRankings } from "./MinervaMenuRenderInfo.js";
+import { filterAvailableStoreStickers, renderCurrentStudentInfo, renderStudentRankings } from "./MinervaMenuRenderInfo.js";
 
 function transitionToMainMenu(studentName) {
+    if (!studentName.status) {
+        chrome.tabs.create({url: chrome.extension.getURL("StudentRegistration.html")});
+
+        return;
+    }
     $("#minervaMenuCollapseCard").collapse("show");
     $("#studentCollapseCard").collapse("hide");
     let studentNamePanel = document.getElementById("studentName-youPanel");
-    studentNamePanel.innerText = "Hi " + studentName;
+    studentNamePanel.innerText = "Hi " + studentName.name;
 }
 
-function displayClassRankings() {
+async function displayClassRankings() {
     let classRankingBody = document.getElementById("class-ranking-body");
-    chrome.runtime.sendMessage({
-        type: "FetchStudentRankings"
-    }, (response) => {
-        if (response.status) {
-            renderStudentRankings(classRankingBody, response.studentInfo);
-        } else {
-            let navRanking = document.getElementById("nav-ranking");
-            navRanking.innerHTML = "No students";
-        }
+    let result = await getStudentRankingsPromise();
+    
+    if (result.status) {
+        renderStudentRankings(classRankingBody, result.studentInfo);
```

Comment:
this is good. tho I would argue that `renderStudentRankings` should query `docuement` to find `class-ranking-body` instead of passing from this function. Queue for change in next PR

Reply:
similar to how you do it in `renderCurrentStudentInfo`